sdk/testing/stepwise/stepwise.go

```diff
@@ -0,0 +1,351 @@
+	MountPath() string
+
+	// BarrierKeys returns the keys used to seal/unseal the cluster. Used for
+	// debugging. TODO verify we should provide this
+	//BarrierKeys()        [][]byte
+
+	// RootToken returns the root token of the cluster, used for making requests
+	// as well as administrative tasks
+	RootToken() string
+}
+
+// PluginType defines the types of plugins supported
+// This type re-create constants as a convienence so users don't need to import/use
+// the consts package.
+type PluginType consts.PluginType
+
+// These are originally defined in sdk/helper/consts/plugin_types.go
+const (
+	PluginTypeUnknown PluginType = iota
+	PluginTypeCredential
+	PluginTypeDatabase
+	PluginTypeSecrets
+)
+
+func (p PluginType) String() string {
+	switch p {
+	case PluginTypeUnknown:
+		return "unknown"
+	case PluginTypeCredential:
+		return "auth"
+	case PluginTypeDatabase:
+		return "database"
+	case PluginTypeSecrets:
+		return "secret"
+	default:
+		return "unsupported"
+	}
+}
+
+// EnvironmentOptions are a collection of options each step driver should
+// support
+type EnvironmentOptions struct {
+	// MountPathPrefix is an optional prefix to use when mounting the plugin. If
+	// omitted the mount path will default to the PluginName with a random suffix.
+	MountPathPrefix string
+
+	// Name is used to register the plugin. This can be arbitray but should be a
+	// reasonable value. For an example, if the plugin in test is a secret backend
+	// that generates UUIDs with the name "vault-plugin-secrets-uuid", then "uuid"
+	// or "test-uuid" would be reasonable. The name is used for lookups in the
+	// catalog. See "name" in the "Register Plugin" endpoint docs:
+	// - https://www.vaultproject.io/api-docs/system/plugins-catalog#register-plugin
+	Name string
+
+	// PluginType is the optional type of plugin. See PluginType const defined
+	// above
+	PluginType PluginType
+
+	// PluginName represents the name of the plugin that gets compiled. In the
+	// standard plugin project file layout, it represents the folder under the
+	// cmd/ folder. In the below example UUID project, the PluginName would be
+	// "uuid":
+	//
+	// vault-plugin-secrets-uuid/
+	// - backend.go
+	// - cmd/
+	// ----uuid/
+	// ------main.go
+	// - path_generate.go
+	//
+	PluginName string
+}
+
+// Step represents a single step of a test Case
+type Step struct {
+	// Operation defines what action is being taken in this step; write, read,
+	// delete, et. al.
+	Operation StepOperation
+
+	// Path is the localized request path. The mount prefix, namespace, and
+	// optionally "auth" will be automatically added.
+	Path string
+
+	// Arguments to pass in the request. These arguments represent payloads sent
+	// to the API.
+	Data map[string]interface{}
+
+	// Check is a function that is called after this step is executed in order to
+	// test that the step executed successfully. If this is not set, then the next
+	// step will be called
+	Check StepCheckFunc
```

Comment:
What do you think about making this into a slice of checks? Barring that, I can see the value in creating a helper function that merges multiple check functions into one:

```go
func MergeStepCheckFuncs(funcs ...StepCheckFunc) StepCheckFunc {
  return func(secret *api.Secret, actualErr error) error {
    for _, f := range funcs {
      err := f(secret, actualErr)
      if err != nil {
        return err
      }
    }
    return nil
  }
}
```

Reply:
I had meant to add this, the legacy framework has one. I'm deferring this for now, while it's a simple addtion as you've shown, I don't have any immedate tests that use it and I feel we can add it later (and add tests for it)